trie/trie.go

```diff
@@ -668,14 +669,15 @@ func (t *Trie) insertRecursive(origNode node, key []byte, pos int, value node) (
 			if len(key) == pos+1 {
 				n.Children[key[pos]] = value
 			} else {
-				n.Children[key[pos]] = &shortNode{Key: common.CopyBytes(key[pos+1:]), Val: value}
+				n.Children[key[pos]] = &shortNode{Key: common.CopyBytes(key[pos+1:]), Val: value, witnessLength: value.witnessLen()}
 			}
 			updated = true
 			n.ref.len = 0
 		} else {
 			updated, nn = t.insertRecursive(child, key, pos+1, value)
 			if updated {
 				t.evictNodeFromHashMap(n)
+				n.witnessLength += nn.witnessLen() - n.Children[key[pos]].witnessLen()
```

Comment:
Why subtraction here?

Reply:
- This code replaces `n.Children[key[pos]]` node by new node. 
- parent.witnessLength contains witnessLen of whole subtree (including self). 
- If we replace 1 child - then need subtract it's len from parent's n.witnessLength of old child (subtree) and add witnessLen of new child (subtree). 
I will add comment about it